authorizer/telegraf.go

```diff
@@ -0,0 +1,152 @@
+	if err := IsAllowed(ctx, *p); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// FindTelegrafConfigByID checks to see if the authorizer on context has read access to the id provided.
+func (s *TelegrafConfigService) FindTelegrafConfigByID(ctx context.Context, id influxdb.ID) (*influxdb.TelegrafConfig, error) {
+	tc, err := s.s.FindTelegrafConfigByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := authorizeReadTelegraf(ctx, tc.OrganizationID, id); err != nil {
+		return nil, err
+	}
+
+	return tc, nil
+}
+
+// FindTelegrafConfig retrieves the telegraf config and checks to see if the authorizer on context has read access to the telegraf config.
+func (s *TelegrafConfigService) FindTelegrafConfig(ctx context.Context, filter influxdb.TelegrafConfigFilter) (*influxdb.TelegrafConfig, error) {
+	tc, err := s.s.FindTelegrafConfig(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := authorizeReadTelegraf(ctx, tc.OrganizationID, tc.ID); err != nil {
+		return nil, err
+	}
+
+	return tc, nil
+}
+
+// FindTelegrafConfigs retrieves all telegraf configs that match the provided filter and then filters the list down to only the resources that are authorized.
+func (s *TelegrafConfigService) FindTelegrafConfigs(ctx context.Context, filter influxdb.TelegrafConfigFilter, opt ...influxdb.FindOptions) ([]*influxdb.TelegrafConfig, int, error) {
+	// TODO: we'll likely want to push this operation into the database eventually since fetching the whole list of data
+	// will likely be expensive.
+	ts, _, err := s.s.FindTelegrafConfigs(ctx, filter, opt...)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	// This filters without allocating
+	// https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating
+	telegrafs := ts[:0]
+	for _, tc := range ts {
+		err := authorizeReadTelegraf(ctx, tc.OrganizationID, tc.ID)
+		if err != nil && influxdb.ErrorCode(err) != influxdb.EUnauthorized {
+			return nil, 0, err
+		}
+
+		if influxdb.ErrorCode(err) == influxdb.EUnauthorized {
+			continue
+		}
+
+		telegrafs = append(telegrafs, tc)
+	}
+
+	return telegrafs, len(telegrafs), nil
+}
+
+// CreateTelegrafConfig checks to see if the authorizer on context has write access to the global telegraf config resource.
+func (s *TelegrafConfigService) CreateTelegrafConfig(ctx context.Context, tc *influxdb.TelegrafConfig, userID influxdb.ID) error {
+	p, err := influxdb.NewPermission(influxdb.WriteAction, influxdb.TelegrafsResourceType, tc.OrganizationID)
+	if err != nil {
+		return err
+	}
+
+	if err := IsAllowed(ctx, *p); err != nil {
+		return err
+	}
+
+	return s.s.CreateTelegrafConfig(ctx, tc, userID)
+}
+
+// UpdateTelegrafConfig checks to see if the authorizer on context has write access to the telegraf config provided.
+func (s *TelegrafConfigService) UpdateTelegrafConfig(ctx context.Context, id influxdb.ID, upd *influxdb.TelegrafConfig, userID influxdb.ID) (*influxdb.TelegrafConfig, error) {
+	tc, err := s.FindTelegrafConfigByID(ctx, id)
```

Comment:
One quick note, this call should be to `s.s`. It likely won't cause an issue for people using the UI, but if people are using tokens for this it'll be problematic.